### PR TITLE
Handling a reboot of a remote node through net-ssh, plus fixing scp command

### PR DESCRIPTION
**Pull request by srbarrios**
## What does this PR change?

Handling a reboot of a remote node through net-ssh, plus fixing scp command

## GUI diff

No difference.

- [x] **DONE**

## Documentation
- No documentation needed

- [x] **DONE**

## Test coverage
ℹ️ If a major new functionality is added, it is **strongly recommended** that tests for the new functionality are added to the Cucumber test suite

- Cucumber tests were refactored

- [x] **DONE**

## Links

No ports

- [x] **DONE**

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [x] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

# Before you merge

Check [How to branch and merge properly](https://github.com/uyuni-project/uyuni/wiki/How-to-branch-and-merge-properly)!

### CODE REVIEW

File: testsuite/features/support/commonlib.rb [hunk 1]
```diff
@@ -617,7 +617,7 @@ def channel_is_synced(channel)
 # @return [ApiTestXmlrpc, ApiTestHttp] The created API client.
 def new_api_client
   ssl_verify = !$is_gh_validation
-  if product == 'SUSE Manager'
+  if $debug_mode || product == 'SUSE Manager'
```
Comment (nodeg) [inline testsuite/features/support/commonlib.rb:620]: With this added again, we will not be able to use the HTTP API with debug mode, which means no debug mode on Uyuni.

**Review by nodeg — APPROVED** [review]

**Review by Bischoff — APPROVED** [review]

File: testsuite/features/support/commonlib.rb (hunk 1 above)
Comment (srbarrios) [reply testsuite/features/support/commonlib.rb:620]: Sorry, a leftover, without it, I can't runin debug mode on my IDE in MacOS due to certificate issue with SSL.
Personally, I would like to keep as it is, but ok, I will remove from this PR, it was a leftover.

Comment (nodeg) [reply testsuite/features/support/commonlib.rb:620]: Sure, fine with me for now :+1:

File: testsuite/features/support/remote_node.rb [hunk 2]
```diff
@@ -266,7 +271,7 @@ def folder_delete(folder)
   #
   # @return [Boolean] true if the node is offline, false otherwise.
   def node_offline?
-    result = run_local('echo test', timeout: 0, check_errors: false).first
```
Comment (maximenoel8) [inline testsuite/features/support/remote_node.rb:269]: Why did you remove this timeout: 0 ? I suppose it's to make the call instant.

**Review by maximenoel8 — APPROVED** [review]
Other than my comment looks good. Thanks Oscar !

File: testsuite/features/support/remote_node.rb (hunk 2 above)
Comment (srbarrios) [reply testsuite/features/support/remote_node.rb:269]: On net-ssh if we set it to zero, it gets stuck, I could not find the proper reason for this, but I suspect the timeout is not applied on the same case, in net-ssh it's described as "how long to wait for the initial connection to be made".
